engine/engine.go

```diff
@@ -553,7 +553,15 @@ func (q *compatibilityQuery) Stats() *stats.Statistics {
 	if q.opts != nil {
 		enablePerStepStats = q.opts.EnablePerStepStats()
 	}
-	return &stats.Statistics{Timers: stats.NewQueryTimers(), Samples: stats.NewQuerySamples(enablePerStepStats)}
+
+	analysis := q.Analyze()
```

Comment:
Don't we return `NoopTelemetry` when analysis is disabled?

Reply:
The `Analyze()` method is currently checking if the root node is observable and returning nil if it is not. So we need the nil check here.